packages/app/src/app/overmind/namespaces/live/internalActions.ts

```diff
@@ -106,16 +110,28 @@ export const initializeModuleState: Action<any> = (
         return;
       }
 
-      module.savedCode = moduleInfo.saved_code;
-      module.code = moduleInfo.code;
-
-      if (moduleInfo.synced) {
-        effects.vscode.sandboxFsSync.writeFile(
-          state.editor.modulesByPath,
-          module
-        );
-      } else {
-        effects.vscode.setModuleCode(module);
+      const savedCodeChanged =
+        getSavedCode(moduleInfo.code, moduleInfo.saved_code) !==
+        getSavedCode(module.code, module.savedCode);
+      const moduleChanged =
+        moduleInfo.code !== module.code ||
+        moduleInfo.saved_code !== module.savedCode;
+
+      if (moduleChanged) {
+        module.savedCode = moduleInfo.saved_code;
+        module.code = moduleInfo.code;
+
+        if (savedCodeChanged) {
+          effects.vscode.sandboxFsSync.writeFile(
```

Comment:
This writes the `module.code` to the filesystem. We only want to do this if the `module.code` is the saved code, AKA the module is `synced`, AKA the `savedCode` is `null`?

Reply:
Hmm, you're right. But what if savedCode has changed but the module is dirty? Maybe we need to change the writeFS call to use getSavedCode as well...